### PR TITLE
[chore](columns) remove update_hashes_with_value for SipHash

### PR DESCRIPTION
**Pull request by wyxxxcat**
## Proposed changes

Issue Number: #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/columns/column_array.cpp [hunk 1]
```diff
@@ -291,11 +291,6 @@ void ColumnArray::update_hash_with_value(size_t n, SipHash& hash) const {
     for (size_t i = 0; i < array_size; ++i) get_data().update_hash_with_value(offset + i, hash);
 }
 
-void ColumnArray::update_hashes_with_value(std::vector<SipHash>& hashes,
-                                           const uint8_t* __restrict null_data) const {
-    SIP_HASHES_FUNCTION_COLUMN_IMPL();
```
Comment (zclllyybb) [inline be/src/vec/columns/column_array.cpp:296]: check if it's safe to delete this

**Review by zclllyybb — APPROVED** [review]
LGTM

**Comment by zclllyybb** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]

**Comment by wyxxxcat** [comment]
run buildall

**Comment by wyxxxcat** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]

**Comment by yiguolei** [comment]
run buildall

File: be/src/vec/columns/column.h [hunk 2]
```diff
@@ -366,19 +353,11 @@ class IColumn : public COW<IColumn> {
...
-    virtual void update_hashes_with_value(std::vector<SipHash>& hashes,
-                                          const uint8_t* __restrict null_data = nullptr) const {
-        LOG(FATAL) << get_name() << " update_hashes_with_value siphash not supported";
-    }
 
     /// Update state of hash function with value of n elements to avoid the virtual function call
     /// null_data to mark whether need to do hash compute, null_data == nullptr
     /// means all element need to do hash function, else only *null_data != 0 need to do hash func
     /// do xxHash here, faster than other sip hash
-    virtual void update_hashes_with_value(uint64_t* __restrict hashes,
```
Comment (zclllyybb) [inline be/src/vec/columns/column.h:378]: Don't delete this

**Review by zclllyybb — CHANGES_REQUESTED** [review]

File: be/test/vec/columns/column_hash_func_test.cpp [hunk 3]
```diff
@@ -113,157 +84,4 @@ TEST(HashFuncTest, ArraySimpleBenchmarkTest) {
...
-    std::vector<uint64_t> xx_hash_vals(3);
-    std::vector<uint32_t> crc_hash_vals(3);
-    auto* __restrict sip_hashes = sip_hash_vals.data();
-    auto* __restrict xx_hashes = xx_hash_vals.data();
-    auto* __restrict crc_hashes = crc_hash_vals.data();
-
-    // sipHash
-    std::vector<SipHash> siphashs(3);
-    array_mutable_col->update_hashes_with_value(siphashs);
-    EXPECT_NO_FATAL_FAILURE(array_mutable_col->update_hashes_with_value(siphashs));
```
Comment (zclllyybb) [inline be/test/vec/columns/column_hash_func_test.cpp:197]: don't delete the whole testcase. only delete sipHash related

**Comment by wyxxxcat** [comment]
run buildall

**Review by zclllyybb — APPROVED** [review]
LGTM

File: be/test/vec/columns/column_hash_func_test.cpp (hunk 3 above)
Comment (zclllyybb) [reply be/test/vec/columns/column_hash_func_test.cpp:197]: this

File: be/src/vec/columns/column.h [hunk 4]
```diff
@@ -395,6 +387,10 @@ class IColumn : public COW<IColumn> {
         LOG(FATAL) << get_name() << "update_crcs_with_value not supported";
     }
 
+    virtual void update_hashes_with_value(uint64_t* __restrict hashes,
```
Comment (zclllyybb) [inline be/src/vec/columns/column.h:390]: dont move them

**Review by zclllyybb — CHANGES_REQUESTED** [review]

**Comment by wyxxxcat** [comment]
run buildall

**Comment by wyxxxcat** [comment]
run buildall

**Comment by wyxxxcat** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]

**Review by yiguolei — DISMISSED** [review]

**Comment by wyxxxcat** [comment]
run buildall

**Comment by wyxxxcat** [comment]
run buildall

**Comment by wyxxxcat** [comment]
run buildall

**Pull request closed without merging**